Packs/CommonScripts/Scripts/ReadFile/ReadFile.py

```diff
@@ -9,22 +11,60 @@ def read_file(args):
         max_file_size = int(max_file_size)
     else:
         max_file_size = 1024 ** 2
-    res = demisto.executeCommand('getFilePath', {'id': args.get('entryID')})
-    file_path = res[0]['Contents']['path']
 
-    with open(file_path, 'r') as f:
+    entry_id = args.get('entryID')
+    input_encoding = args.get('input_encoding') or None
+    output_data_type = args.get('output_data_type') or 'raw'
+    output_meta_data = argToBoolean(args.get('output_meta_data') or 'false')
+
+    file_path = execute_command('getFilePath', {'id': entry_id})['path']
+    file_size = os.path.getsize(file_path)
+
+    with open(file_path,
+              'rb' if input_encoding == 'binary' else 'r',
+              encoding=None if input_encoding == 'binary' else input_encoding) as f:
         data = f.read(max_file_size)
+        eof = len(f.read(1)) == 0
 
-    if data:
-        message = 'Read {} bytes from file.'.format(len(data))
-        result = {"Type": entryTypes["note"],
-                  "ContentsFormat": formats["text"],
-                  "Contents": {"FileData": data},
-                  "HumanReadable": message,
-                  "EntryContext": {"FileData": data}
-                  }
-    else:
+    if not output_meta_data and len(data) == 0:
         raise Exception('No data could be read.')
+
+    if isinstance(data, bytes):
+        message = f'Read {len(data)} bytes from file.'
+    else:
+        message = f'Read {len(data)} charactors from file.'
```

Comment:
Please revert this change and keep the original message this can break backwards compatibility.